src/Aspire.Hosting/Utils/VolumeNameGenerator.cs

```diff
@@ -5,16 +5,26 @@
 
 namespace Aspire.Hosting.Utils;
 
-internal static class VolumeNameGenerator
+/// <summary>
+/// Utility class for generating volume names
+/// </summary>
+public static class VolumeNameGenerator
 {
+    /// <summary>
+    /// Creates a volume name with the form <c>{applicationName}-{sha256 of apphost path}-{resourceName}-{suffix}</c>, e.g. <c>myapplication-a345f2451-postgres-data</c>.
+    /// </summary>
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="suffix">The suffix to append to the volume name.</param>
+    /// <returns>The volume name.</returns>
+    /// <exception cref="ArgumentException"></exception>
     public static string CreateVolumeName<T>(IResourceBuilder<T> builder, string suffix) where T : IResource
     {
         if (!HasOnlyValidChars(suffix))
         {
             throw new ArgumentException($"The suffix '{suffix}' contains invalid characters. Only [a-zA-Z0-9_.-] are allowed.", nameof(suffix));
```

Comment:
```suggestion
            throw new ArgumentException($"The suffix '{suffix}' contains invalid characters. It must match [a-zA-Z0-9][a-zA-Z0-9_.-]*.", nameof(suffix));
```
perhaps?

Reply:
I like that, thanks!

Reply:
I don't see this change for some reason.